onnxruntime/core/providers/cuda/tensor/sequence_op.cc

```diff
@@ -12,6 +12,7 @@ ONNX_OPERATOR_KERNEL_EX(
     11,
     kCudaExecutionProvider,
     (*KernelDefBuilder::Create())
+        .InputMemoryType(OrtMemTypeCPUInput, 1)
```

Comment:
Without this a copy node to CUDA is bound to be inserted (most likely because these are produced by ops writing to CPU e.g.: Shape)